grammar/grammar.y

```diff
@@ -408,118 +413,98 @@ regexp
 boolean_expression
     : expression
       {
-        
+        $$ = $1
       }
     ;
 
 expression
     : _TRUE_
       {
-          
+        $$ = data.Expression{Left:true}
       }
     | _FALSE_
       {
-        
+        $$ = data.Expression{Left:false}
       }
     | primary_expression _MATCHES_ regexp
       {
-        
+        $$ = data.Expression{Left: $1, Operator: "matches", Right: $<reg>3}
       }
     | primary_expression _CONTAINS_ primary_expression
       {
-        
+        $$ = data.Expression{Unsupported: true}
       }
     | _STRING_IDENTIFIER_
       {
-        
+        $$ = data.Expression{Unsupported: true}
```

Comment:
I'm trying to think of a good way of being able to refer back to the `data.String` struct that this _STRING_IDENTIFIER_ corresponds to.  Any ideas using the `data.Expression` struct?  Possibly embed a pointer to a `data.String`?